lib/servers/iXonServer/AndorVideo.py

```diff
@@ -210,10 +211,12 @@ def live_update(self):
         self.img_view.setImage(image_data.transpose(), autoRange = False, autoLevels = False, pos = [self.startx, self.starty], scale = [self.binx,self.biny], autoHistogramRange = False)
 
         if self.save_images_state == True:
-            dt = datetime.now()
-            time_stamp = str(dt.year)+str(dt.month)+str(dt.day)+str(dt.hour)\
-            +str(dt.minute)+str(dt.second)+str(dt.microsecond)+'.csv'
-            np.savetxt(self.image_path+time_stamp,image_data)
+            if not np.array_equal(image_data, self.saved_data):
+                self.saved_data = image_data
+                dt = datetime.now()
+                time_stamp = str(dt.year).rjust(4,"0")+str(dt.month).rjust(2,"0")+str(dt.day).rjust(2,"0")+str(dt.hour).rjust(2,"0")\
```

Comment:
It might be good to break this up over two or three lines.  It looks long.  And then add all the strings together on the time_stamp line.